### DISCUSSION TITLE
Is it possible to check empty input values and change them with default within the ${} expression?

### DISCUSSION THREAD

**Discussion by AliakseiBuhakou** [Netflix/conductor#2878] [Q&A]
Hello. Could you please guide me with following question.
There is an workflow example.
`{
			"name": "task1",
			"taskReferenceName": "task1",
			"inputParameters": {
				"http_request": {
					"uri": "some_uri",
					"method": "POST",
					"body": { some_body
					},
					"connectionTimeOut": "600000",
					"readTimeOut": "600000",
					"headers": {
						"header1": "${workflow.input.header1}"
					}
				}
			}`

The value for "header1": "${workflow.input.header1}" is optional. Is it possible to check the value and change it with some default string? It can be solved with some JSON_JQ_TRANSFORM probably, is there a way to do this without lambda? The reason of the question is that on Conductor v3+ it fails with NPE if value is not provided, Conductor v2+ works fine

**Comment by apanicker-nflx** [comment]
There isn't an implicit way to do this currently. You would need to do this validation in an INLINE task preceding the HTTP task.

**Reply by Thomas-Kotak** [reply]
@apanicker-nflx if we make a Pull Request for this, is it likely to get merged? Our project needs to handle this case so we are trying to decide whether to do this via Lambda function or contributing to the project.

**Reply by apanicker-nflx** [reply]
Yes, I would recommend submitting a proposal for the implementation plan and then proceeding with the PR so that we reach an agreed upon implementation sooner. 
Additionally, you could take a look at the use of `inputTemplate` in task definition which achieves this functionality to some extent.
